h/services/bulk_api/group.py

```diff
@@ -16,8 +16,8 @@ class BulkGroup:
 class BulkGroupService:
     """A service for retrieving groups in bulk."""
 
-    def __init__(self, db: Session):
-        self._db = db
+    def __init__(self, db_replica: Session):
```

Comment:
If this is successfully I could see many services using both db and db_replica.

We only make one query here, no need to have references to both.

Reply:
👍 - I think it is important to make it super clear whether code expects to be working with the normal DB connection, or can work with the replica.